Installation fails with Helm 2.16.9: `can't give argument to non-function hasKey .Values "global"`

Hi, I just cloned the repository (cc2697b7f1f550d9e8983f5c922141388472d54b) and ran `helm dependency update` followed by `helm install . --name st2` and the installation fails. Running `helm lint`:

```
sharmaabhyudaya@cloudshell:~/stackstorm-ha$ helm lint
==> Linting .
[ERROR] templates/: render error in "stackstorm-ha/charts/mongodb-replicaset/templates/tests/mongodb-up-test-pod.yaml": template: stackstorm-ha/charts/mongodb-replicaset/templates/_helpers.tpl:84:14: executing "mongodb-replicaset.namespace" at <(hasKey .Values "global") .Values.global.namespaceOverride>: can't give argument to non-function hasKey .Values "global"
```

It looks like a problem with the [mongo-replicaset helm chart](https://github.com/helm/charts/tree/master/stable/mongodb-replicaset) and this particular code path was introduced with https://github.com/helm/charts/pull/21203 . I'm not sure if this is an issue with the Stackstorm helm chart or with Helm or the mongo-replicaset helm chart.

```
$ helm version
Client: &version.Version{SemVer:"v2.16.9", GitCommit:"8ad7037828e5a0fca1009dabe290130da6368e39", GitTreeState:"clean"}
```

Interestingly, the chart install successfully with Helm v2.16.1. I haven't tried other versions but we can possibly experiment where it all starts falling apart.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -17,7 +17,7 @@ jobs:
     working_directory: ~/stackstorm-ha
     docker:
       # Pin Helm to v2.x, see https://github.com/StackStorm/stackstorm-ha/issues/98
-      - image: lachlanevenson/k8s-helm:v2.16.7
+      - image: lachlanevenson/k8s-helm:v2.16.9
     steps:
       - checkout
       - run:

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,11 +1,9 @@
 # Changelog
 
-## In Development
-* Allow injection of datastore key in cluster (#115) (by @AngryDeveloper)
-
 ## v0.31.0
 
 * Makes the chart compatible with Helm versions >= `2.16.8` by downgrading `mongodb-replicaset` from `3.14.0` to `3.12.0` (#137) (by @AbhyudayaSharma)
+* Allow injection of datastore key in cluster (#115) (by @AngryDeveloper)
 
 ## v0.30.0
 * Pin st2 version to `v3.3dev` as a new latest development version (#129)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,8 +1,7 @@
 # Changelog
 
 ## v0.31.0
-
-* Makes the chart compatible with Helm versions >= `2.16.8` by downgrading `mongodb-replicaset` from `3.14.0` to `3.12.0` (#137) (by @AbhyudayaSharma)
+* Fix chart compatibility with Helm versions >= `2.16.8` by downgrading `mongodb-replicaset` from `3.14.0` to `3.12.0` (#137) (by @AbhyudayaSharma)
 * Allow injection of datastore key in cluster (#115) (by @AngryDeveloper)
 
 ## v0.30.0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,10 @@
 ## In Development
 * Allow injection of datastore key in cluster (#115) (by @AngryDeveloper)
 
+## v0.31.0
+
+* Makes the chart compatible with Helm versions >= `2.16.8` by downgrading `mongodb-replicaset` from `3.14.0` to `3.12.0` (#137) (by @AbhyudayaSharma)
+
 ## v0.30.0
 * Pin st2 version to `v3.3dev` as a new latest development version (#129)
 * Migrate from `py2` `Ubuntu Xenial` to `py3` `Ubuntu Bionic` as a base StackStorm OS (StackStorm/st2-dockerfiles#16, #129)

--- a/Chart.yaml
+++ b/Chart.yaml
@@ -2,7 +2,7 @@ apiVersion: v1
 # Update StackStorm version here to rely on other Docker images tags
 appVersion: 3.3dev
 name: stackstorm-ha
-version: 0.30.0
+version: 0.31.0
 description: StackStorm K8s Helm Chart, optimized for running StackStorm in HA environment.
 home: https://stackstorm.com/#product
 icon: https://avatars1.githubusercontent.com/u/4969009

--- a/requirements.yaml
+++ b/requirements.yaml
@@ -4,7 +4,7 @@ dependencies:
     repository: https://kubernetes-charts.storage.googleapis.com/
     condition: rabbitmq-ha.enabled
   - name: mongodb-replicaset
-    version: 3.14.0
+    version: 3.12.0
     repository: https://kubernetes-charts.storage.googleapis.com/
     alias: mongodb-ha
     condition: mongodb-ha.enabled

